packages/pds/src/config.ts

```diff
@@ -82,16 +90,26 @@ export class ServerConfig {
       jwtSecret,
       recoveryKey,
       didPlcUrl,
+      serverDid,
       adminPassword,
       inviteRequired,
       blockstoreLocation,
       databaseLocation,
-      testNameRegistry,
+      availableUserDomains,
       appUrlPasswordReset,
       emailSmtpUrl,
       emailNoReplyAddress,
       ...overrides,
     })
+
+    const invalidDomain = cfg.availableUserDomains.find(
+      (domain) => domain.length > 0 && domain.startsWith('.'),
+    )
+    if (invalidDomain) {
+      throw new Error(`Invalid domain: ${invalidDomain}`)
+    }
```

Comment:
What if this validation happened in `ServerConfig`'s constructor?  The only upside there is that no matter how you create the `ServerConfig`, it will be validated.

Sort of related, I think that the tests `availableUserDomains: ['.test']` would fail this validation though we want it to succeed.  Perhaps we want something like this:
```suggestion
    const invalidDomain = cfg.availableUserDomains.find(
      (domain) => !domain.startsWith('.'),
    )
    if (invalidDomain) {
      throw new Error(`Invalid domain: ${invalidDomain}`)
    }
```